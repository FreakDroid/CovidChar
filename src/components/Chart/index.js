import {Bar, BarChart, CartesianGrid, Legend, Tooltip, XAxis, YAxis} from 'recharts';
import React from 'react';
import PropTypes from 'prop-types';


export default function Chart({data, dataKeyXAxis, dataKeyBar, dataKeyBarSecond}) {
  return (
    <BarChart
      width={1300}
      height={700}
      data={data}
      margin={{
        top: 5,
        right: 30,
        left: 40,
        bottom: 5
      }}
    >
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey={dataKeyXAxis} />
      <YAxis />
      <Tooltip />
      <Legend />
      <Bar dataKey={dataKeyBar} fill="#8884d8" />
      <Bar dataKey={dataKeyBarSecond} fill="#82ca9d" />
    </BarChart>
  )
}

Chart.propTypes = {
  data: PropTypes.array.isRequired,
  dataKeyXAxis: PropTypes.string,
  dataKeyBar: PropTypes.string,
  dataKeyBarSecond: PropTypes.string
}