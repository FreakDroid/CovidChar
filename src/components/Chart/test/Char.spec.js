import {render} from '@testing-library/react';
import Chart from '../index';

describe('Chart Component', () => {

  const props = {
    data: [],
    dataKeyBar: "cases",
    dataKeyXAxis: 'number',
    dataKeyBarSecond: 'deaths'
  }

  const component = () => render(<Chart {...props}/>);

  it('Should render the component properly', () => {
    const { baseElement } = component();
    expect(baseElement).toMatchSnapshot();
  });
});