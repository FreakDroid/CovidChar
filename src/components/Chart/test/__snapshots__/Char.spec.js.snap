// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Component Should render the component properly 1`] = `
<body>
  <div>
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 1300px; height: 700px;"
    >
      <svg
        class="recharts-surface"
        height="700"
        version="1.1"
        viewBox="0 0 1300 700"
        width="1300"
      >
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="660"
              width="1170"
              x="100"
              y="5"
            />
          </clippath>
        </defs>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="660"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="1170"
              x="100"
              x1="100"
              x2="1270"
              y="5"
              y1="5"
              y2="5"
            />
            <line
              fill="none"
              height="660"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="1170"
              x="100"
              x1="100"
              x2="1270"
              y="5"
              y1="665"
              y2="665"
            />
          </g>
          <g
            class="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height="660"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="1170"
              x="100"
              x1="392.5"
              x2="392.5"
              y="5"
              y1="5"
              y2="665"
            />
            <line
              fill="none"
              height="660"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="1170"
              x="100"
              x1="977.5"
              x2="977.5"
              y="5"
              y1="5"
              y2="665"
            />
            <line
              fill="none"
              height="660"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="1170"
              x="100"
              x1="100"
              x2="100"
              y="5"
              y1="5"
              y2="665"
            />
            <line
              fill="none"
              height="660"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="1170"
              x="100"
              x1="1270"
              x2="1270"
              y="5"
              y1="5"
              y2="665"
            />
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="30"
            orientation="bottom"
            stroke="#666"
            type="category"
            width="1170"
            x="100"
            x1="100"
            x2="1270"
            y="665"
            y1="665"
            y2="665"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="1170"
                x="100"
                x1="392.5"
                x2="392.5"
                y="665"
                y1="671"
                y2="665"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                type="category"
                width="1170"
                x="392.5"
                y="673"
              >
                <tspan
                  dy="0.71em"
                  x="392.5"
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="1170"
                x="100"
                x1="977.5"
                x2="977.5"
                y="665"
                y1="671"
                y2="665"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                type="category"
                width="1170"
                x="977.5"
                y="673"
              >
                <tspan
                  dy="0.71em"
                  x="977.5"
                >
                  auto
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="660"
            orientation="left"
            stroke="#666"
            type="number"
            width="60"
            x="40"
            x1="100"
            x2="100"
            y="5"
            y1="5"
            y2="665"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          />
        </g>
      </svg>
      <div
        class="recharts-legend-wrapper"
        style="position: absolute; width: 1230px; height: auto; left: 40px; bottom: 5px;"
      >
        <ul
          class="recharts-default-legend"
          style="padding: 0px; margin: 0px; text-align: center;"
        >
          <li
            class="recharts-legend-item legend-item-0"
            style="display: inline-block; margin-right: 10px;"
          >
            <svg
              class="recharts-surface"
              height="14"
              style="display: inline-block; vertical-align: middle; margin-right: 4px;"
              version="1.1"
              viewBox="0 0 32 32"
              width="14"
            >
              <path
                class="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#8884d8"
                stroke="none"
              />
            </svg>
            <span
              class="recharts-legend-item-text"
              style="color: rgb(136, 132, 216);"
            >
              cases
            </span>
          </li>
          <li
            class="recharts-legend-item legend-item-1"
            style="display: inline-block; margin-right: 10px;"
          >
            <svg
              class="recharts-surface"
              height="14"
              style="display: inline-block; vertical-align: middle; margin-right: 4px;"
              version="1.1"
              viewBox="0 0 32 32"
              width="14"
            >
              <path
                class="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#82ca9d"
                stroke="none"
              />
            </svg>
            <span
              class="recharts-legend-item-text"
              style="color: rgb(130, 202, 157);"
            >
              deaths
            </span>
          </li>
        </ul>
      </div>
      <div
        class="recharts-tooltip-wrapper"
        style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="recharts-default-tooltip"
          style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
        >
          <p
            class="recharts-tooltip-label"
            style="margin: 0px;"
          />
        </div>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    id="recharts_measurement_span"
    style="position: absolute; top: -20000px; left: 0px; padding: 0px; margin: 0px; white-space: pre;"
  >
    Â 
  </span>
</body>
`;
